### PR TITLE
LIVY-230. SparkSession is not avalible for PySparkInterperter and SparkRInterperter

### PR DESCRIPTION
**Pull request by zjffdu**
This PR make SparkSession avalible in PySparkInterpreter and SparkRInterpreter.

### CODE REVIEW

**Comment by tc0312** [comment]
LGTM. Do you mind adding a test for it?

**Comment by jerryshao** [comment]
@linchan-ms adding Spark2 supports for `JobContextImpl` is still in consideration, it is not so straightforward compared to interpreter things.

File: repl/src/main/resources/fake_shell.py [hunk 1]
```diff
@@ -523,6 +524,8 @@ def main():
             exec('from pyspark.sql import HiveContext', global_dict)
```
Comment (tc0312) [inline repl/src/main/resources/fake_shell.py]: I think this is imported for PySpark RSC support.
I think in the long run we should separate repl code and rsc code to 2 separate python files. And we can fix this when we add SparkSession support for RSC.

File: repl/src/main/scala/com/cloudera/livy/repl/SparkRInterpreter.scala [hunk 2]
```diff
@@ -125,9 +126,10 @@ object SparkRInterpreter {
   }
 }
 
-class SparkRInterpreter(process: Process, backendInstance: Any, backendThread: Thread)
-  extends ProcessInterpreter(process)
-{
+class SparkRInterpreter(process: Process,
+                        backendInstance: Any,
```
Comment (zjffdu) [inline repl/src/main/scala/com/cloudera/livy/repl/SparkRInterpreter.scala]: Isn't this style more readable ?

Comment (tc0312) [inline repl/src/main/scala/com/cloudera/livy/repl/SparkRInterpreter.scala]: Personally I have no opinion on this. But existing code follows Spark's coding style. Let's be consistent and follow Spark's coding style.

Comment (zjffdu) [inline repl/src/main/scala/com/cloudera/livy/repl/SparkRInterpreter.scala]: Fixed

**Pull request merged**